### PR TITLE
Detect path when running an upgrade

### LINKED ISSUE #2737: Server upgrade fails
## What went wrong? 😲
 
With the issue #2471 coming in, the upgrade process of open mSupply server has an issue.
The existing server is detected, and the upgrade runs successfully. However, the new version is installed into the folder `Open mSupply Server` and the configuration and logs remain in the previous installation folder (in my case, 'omSupply Server').

As a result, the service does not start but enters an endless starting state.

## Expected behaviour 🤔
I would expect the current folder to be used. 
Please retest this scenario too:

* Install an earlier version into a non-standard folder
* Upgrade

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Install a previous version
2. Upgrade to v1.6.00-rc1

## Your environment 🌱

 - Version: 
 - Platform:
   - [ ] android (tablet)
   - [ ] browser (extra points if you tell us which one)
   - [ ] desktop (windows)
   - [ ] desktop (macOS)
   - [x] server (windows)

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #2737

# 👩🏻‍💻 What does this PR do? 
 
Sets the install path to be the currently installed location, if an upgrade is detected.

# 🧪 How has/should this change been tested? 

- Install a previous version into a non-standard folder. 
- Install this version: you should see the upgrade process and it will detect the current folder

## 📃 Documentation

_No user facing changes_

### CODE REVIEW

File: build/windows/omsupply_server.suf [hunk 1]
```diff
@@ -4404,7 +4409,7 @@ g_HandleSystemReboot();
 </SessionVar>
 <SessionVar>
 <Name>%omsupply_db_name%</Name>
-<Value>omsupply-database</Value>
+<Value>open-msupply</Value>
```
Comment (mark-prins) [inline build/windows/omsupply_server.suf:4412]: not part of the bug fix, just tidying up.

File: build/windows/omsupply_standalone.suf [hunk 2]
```diff
@@ -3492,7 +3497,7 @@ g_HandleSystemReboot();
 </SessionVar>
 <SessionVar>
 <Name>%omsupply_db_name%</Name>
-<Value>open-msupply-database</Value>
+<Value>open-msupply</Value>
```
Comment (mark-prins) [inline build/windows/omsupply_standalone.suf:3500]: made it consistent

File: build/windows/omsupply_server.suf (hunk 1 above)
Comment (clemens-msupply) [reply build/windows/omsupply_server.suf:4412]: I have no idea what it's doing :) but can you rename the database name? e.g. what if it is just a version upgrade?

**Comment by mark-prins** [comment]
> can you rename the database name? e.g. what if it is just a version upgrade?

The database name is only used for new installs - we take the name and patch a `local.yaml` file.. but for upgrades, it doesn't touch the config.

**Review by clemens-msupply — COMMENTED** [review]
Works fine. I installed 1.6 Rc2 and then Rc3 and it picked my previously selected directory.

However, when uninstalling it I got two entries Rc2 + Rc3 after unistalling one of them I can't uninstall the other anymore...

image

This might be an unrelated problem though?

**Comment by mark-prins** [comment]
I think unrelated.. though worth looking at since I'm working on the installer. I only saw the one entry after installing the second.. will check and see how windows determines a unique app

**Review by clemens-msupply — APPROVED** [review]

**Comment by mark-prins** [comment]
I've found the issue with the upgrade - the unique registry key value in the uninstall settings was appending the product version to the product name, which meant that is was creating a new entry in the uninstall list 🙄 

This won't fix the issue of upgrading to 1.6.00+ however, because of #2471
The name has changed from 1.6 onwards and so will always create a new entry. after that though, the upgrade works!

You can test with an upgrade from [v1.5.04-install-test](https://jenkins.msupply.org:8443/job/omSupplyMain%20-%20Installers/202/) to [1.6.00-install-test](https://jenkins.msupply.org:8443/job/omSupplyMain%20-%20Installers/203/)

**Pull request merged**